indico/modules/rb/client/js/common/bookings/BookingDetails.jsx

```diff
@@ -458,20 +506,35 @@ class BookingDetails extends React.Component {
         )}
         {canReject && (
           <Popup trigger={rejectButton} position="bottom center" on="click">
-            <FinalForm onSubmit={data => this.changeState('reject', data)} render={renderForm} />
+            <FinalForm
+              onSubmit={data => this.changeState('reject', data)}
+              render={renderRejectionForm}
+            />
           </Popup>
         )}
         {showAccept && (
-          <Button
-            type="button"
-            icon="check circle"
-            color="green"
-            size="small"
-            onClick={() => this.changeState('approve')}
-            loading={actionInProgress === 'approve' && bookingStateChangeInProgress}
-            disabled={bookingStateChangeInProgress}
-            content={Translate.string('Accept booking')}
-          />
+          <Button.Group size="small" color="green">
+            <Button
+              type="button"
+              icon="check circle"
+              onClick={() => this.changeState('approve')}
+              loading={actionInProgress === 'approve' && bookingStateChangeInProgress}
+              disabled={bookingStateChangeInProgress || activeAcceptanceForm}
+              content={Translate.string('Accept booking')}
+            />
+            <Popup
+              trigger={acceptWithMessage}
+              position="bottom right"
+              on="click"
+              onOpen={this.activateAcceptanceForm}
+              onClose={this.deactivateAcceptanceForm}
+            >
+              <FinalForm
```

Comment:
I think both this form and the rejection one could use a subscription prop to avoid unnecessary re-renders:

    subscription={{hasValidationErrors: true, submitSucceeded: true, submitting: true, pristine: true}}

(you basically list all props you want to receive in the form's render function, and anything else which you don't use anyway won't be passed and thus also won't trigger a re-render)